src/wald_wolfowitz.jl

```diff
@@ -0,0 +1,57 @@
+export 
+    WaldWolfowitzTest
+
+struct WaldWolfowitzTest <: HypothesisTest
+    N_above::Int
+    N_below::Int
+    N_runs::Int
+    μ::Real
+    σ::Real
+    z::Real
+end
+
+testname(::WaldWolfowitzTest) = "Wald-Wolfowitz Test"
+population_param_of_interest(x::WaldWolfowitzTest) = ("Number of runs", x.μ, x.N_runs) # parameter of interest: name, value under h0, point estimate
+default_tail(::WaldWolfowitzTest) = :both
+
+
+function show_params(io::IO, x::WaldWolfowitzTest, ident="")
+    println(io, ident, "number of runs:  $(x.N_runs)")
+    println(io, ident, "z-statistic:     $(x.z)")
+end
+
+"""
+    WaldWolfowitzTest(x::AbstractVector{T<:Real})
+
+Performs the Wald-Wolfowitz (or Runs) test to determine
+randommness for a data-sequence. The null hypothesis is that the data is random, 
```

Comment:
Wikipedia describes the test as applying to two-valued data sequences, so it would be useful to explain that values are split at the median (which as a special case works when there are only two values).